pandas/core/frame.py

```diff
@@ -10596,6 +10596,17 @@ def quantile(
         validate_percentile(q)
         axis = self._get_axis_number(axis)
 
+        if numeric_only is no_default:
```

Comment:
ahh, this is *way* too noisy. we should only show this warning if we *have* non-numeric columns.

Reply:
So I think you would agree with me in this discussion
https://github.com/pandas-dev/pandas/pull/46096#discussion_r817226386

or how else can we do that?

Reply:
you would select the dtypes and if any non-numeric then process the warning (which still won't show if the actual value is specified).

Reply:
Ok, I just thought that the datetime and timedelta were the only non numeric types that made sense. I will try to be more general with testing specific for numeric.